src/react/hooks/paragon/useParagonTheme.js

```diff
@@ -0,0 +1,111 @@
+import {
+  useCallback, useEffect, useReducer, useState,
+} from 'react';
+
+import { SELECTED_THEME_VARIANT_KEY } from '../../constants';
+import { logError } from '../../../logging';
+import { paragonThemeActions, paragonThemeReducer } from '../../reducers';
+import { getDefaultThemeVariant, isEmptyObject } from './utils';
+
+import useParagonThemeCore from './useParagonThemeCore';
+import useParagonThemeUrls from './useParagonThemeUrls';
+import useParagonThemeVariants from './useParagonThemeVariants';
+
+/**
+ * Given the inputs of URLs to the CSS for the core application theme and the theme variants (e.g., light), this hook
+ * will inject the CSS as `<link>` elements into the HTML document, loading each theme variant's CSS with an appropriate
+ * priority based on whether its the currently active theme variant. This is done using "alternate" stylesheets. That
+ * is,the browser will still download the CSS for the non-current theme variants, but at a lower priority than the
+ * current theme variant's CSS. This ensures that if the theme variant is changed at runtime, the CSS for the new theme
+ * variant will already be loaded.
+ *
+ * @memberof module:React
+ * @param {object} config An object containing the URLs for the theme's core CSS and any theme (i.e., `getConfig()`)
+ *
+ * @returns An array containing 2 elements: 1) an object containing the app
+ *  theme state, and 2) a dispatch function to mutate the app theme state.
+ */
+const useParagonTheme = (config) => {
```

Comment:
[nit/consideration] Likely could simplify the usage of this hook slightly, as I believe `getConfig` could be called directly within this hook, where appropriate, versus needing to pass the `config` object as a function parameter.

For example, would it make sense to call `getConfig` within `useParagonThemeUrls` directly instead of passing `config` variable around?

Reply:
Yeah, I think this make sense because the configuration variables tend to be loaded at the first load (ideally once) and not change during the user navigation. I will made the suggested change.